documentation/design-docs/ipc-protocol.md

```diff
@@ -0,0 +1,208 @@
+# IPC Protocol
+
+## Overview
+
+This spec describes a proposed IPC Protocol to be used for communicating with the runtime's diagnostics server from an external client over a platform-specific transport, e.g., Unix Domain Sockets. This protocol should be lightweight and generic, so as to be extensible for future use cases.  The primary use today is for EventPipe.
+
+## TODOs
```

Comment:
I'd suggest moving any sections that you don't plan to have in the final version of the doc to the bottom. That lets you keep it for reference but you can place a clear cut-line about what text is expected in the final doc and what is notes in-progress. In particular sections about TODO or current impl wouldn't be needed in the final.